src/component/playground/Playground/PlaygroundCodeFieldset/PlaygroundCodeFieldset.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+    Dispatch,
+    SetStateAction,
+    useEffect,
+    useMemo,
+    useState,
+    VFC,
+} from 'react';
+import {
+    Box,
+    Button,
+    FormControl,
+    InputLabel,
+    MenuItem,
+    Select,
+    TextField,
+    Typography,
+    useTheme,
+} from '@mui/material';
+import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import { debounce } from 'debounce';
+
+interface IPlaygroundCodeFieldsetProps {
+    value: string | undefined;
+    setValue: Dispatch<SetStateAction<string | undefined>>;
+}
+
+export const PlaygroundCodeFieldset: VFC<IPlaygroundCodeFieldsetProps> = ({
+    value,
+    setValue,
+}) => {
+    const theme = useTheme();
+    const { context } = useUnleashContext();
+    const contextOptions = useMemo(
+        () =>
+            context
+                .sort((a, b) => a.sortOrder - b.sortOrder)
+                .map(({ name }) => name),
+        [context]
+    );
+    const [error, setError] = useState<string>();
+    const debounceSetError = useMemo(
+        () =>
+            debounce((input?: string) => {
+                if (!input) {
+                    return setError(undefined);
+                }
+
+                try {
+                    JSON.parse(input);
+                } catch (error: unknown) {
+                    return setError(formatUnknownError(error));
+                }
+
+                return setError(undefined);
+            }, 250),
+        [setError]
+    );
+
+    useEffect(() => {
+        debounceSetError(value);
+    }, [debounceSetError, value]);
+
+    const [contextField, setContextField] = useState<string>('');
+    const [contextValue, setContextValue] = useState<string>('');
+    const onAddField = () => {
+        try {
+            const currentValue = JSON.parse(value || '{}');
+            setValue(
+                JSON.stringify(
+                    {
+                        ...currentValue,
+                        [contextField]: contextValue,
+                    },
+                    null,
+                    2
+                )
+            );
+        } catch (error) {
+            // Shouldn't happened, because button is disabled when there is an error
```

Comment:
So if I understand this correctly, there will be a delay where the button will be functional before the error is updated. Should we give any feedback if that is the case or is the button being disabled enough feedback?